parse/parser.go

```diff
@@ -484,9 +484,9 @@ func parseIdentifier(buf TokenBuffer) (ast.Expression, error) {
 		return nil, ExpectError{token, Ident}
 	}
 
-	// TODO: check whether variable name exist,
-	// TODO: but do not update symbol table in this case
-	// TODO: if not, return error
+	if exist := scope.Get(token.Val); exist != nil {
```

Comment:
After we `token := buf.Read()` in 482 line, we should check whether this symbol exist in `scope`, if not exist in scope then should return error

Reply:
And plz add tcs for this logic